cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/ZookeeperLeaderFinderTest.java

```diff
@@ -228,94 +217,21 @@ public void test1PodClusterReturnsOnlyPodAsLeader(VertxTestContext context) {
         ZookeeperLeaderFinder finder = new ZookeeperLeaderFinder(vertx, this::backoff);
         Checkpoint a = context.checkpoint();
         int firstPodIndex = 0;
-        finder.findZookeeperLeader(Reconciliation.DUMMY_RECONCILIATION, Set.of(createPodWithId(firstPodIndex)), dummySecret(), dummySecret())
+        finder.findZookeeperLeader(Reconciliation.DUMMY_RECONCILIATION, Set.of(createPodWithId(firstPodIndex)), DUMMY_IDENTITY)
             .onComplete(context.succeeding(leader -> {
                 context.verify(() -> assertThat(leader, is("my-cluster-zookeeper-0")));
                 a.flag();
             }));
     }
 
-    @Test
-    public void testSecretWithMissingClusterOperatorKeyThrowsException(VertxTestContext context) {
-        SecretOperator mock = mock(SecretOperator.class);
-        ZookeeperLeaderFinder finder = new ZookeeperLeaderFinder(vertx, this::backoff);
-
-        Mockito.reset(mock);
-        when(mock.getAsync(eq(NAMESPACE), eq(KafkaResources.clusterCaCertificateSecretName(CLUSTER))))
-            .thenReturn(Future.succeededFuture(
-                    new SecretBuilder()
-                            .withNewMetadata()
-                                .withName(KafkaResources.clusterCaCertificateSecretName(CLUSTER))
-                                .withNamespace(NAMESPACE)
-                            .endMetadata()
-                            .withData(emptyMap())
-                            .build()));
-
-        Secret secretWithMissingClusterOperatorKey = new SecretBuilder()
-                .withNewMetadata()
-                .withName(KafkaResources.secretName(CLUSTER))
-                .withNamespace(NAMESPACE)
-                .endMetadata()
-                .withData(emptyMap())
-                .build();
-
-        Checkpoint a = context.checkpoint();
-
-        finder.findZookeeperLeader(Reconciliation.DUMMY_RECONCILIATION, treeSet(createPodWithId(0), createPodWithId(1)), dummySecret(), secretWithMissingClusterOperatorKey)
-            .onComplete(context.failing(e -> context.verify(() -> {
-                assertThat(e, instanceOf(RuntimeException.class));
-                assertThat(e.getMessage(),
-                        is("The Secret testns/testcluster-cluster-operator-certs is missing the key cluster-operator.key"));
-                a.flag();
-            })));
-
-    }
-
-    @Test
-    public void testSecretsCorrupted(VertxTestContext context) {
```

Comment:
Do we replace these tests on the higher level?

Reply:
Yes, they have been moved to PemAuthIdentityTest, since the logic it is testing has moved to PemAuthIdentity.